### PR TITLE
Introduce locally-defined `Rejection` reasons instead of the participant-state rejection reason type. [KVL-1002]

### PR DESCRIPTION
**Pull request by SamirTalwar-DA**
Instead of using the `RejectionReasonV0` enumeration defined in _participant-state_, we change the code to use new enumerations that are specific to the ledger driver. For each component, we define an enumeration named `Rejection`, encompassing all possible error states, and convert these cases to the _participant-state_ rejection when necessary.

The components in question are the Indexer (both the current indexer and the append-only one), Sandbox Classic, and kvutils.

Defining these enumerations makes the migration to the participant state v2 API a little easier, mostly because we have fewer dependencies, but also because the data is now structured, which will make it easier to construct valid v2 command rejections containing the [gRPC `ErrorInfo` type](https://github.com/googleapis/googleapis/blob/3859723d77717f5082e78b4b0565c6d08b3eedfe/google/rpc/error_details.proto#L88-L136).

Constructing those `ErrorInfo` objects will come in a future changeset.

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [x] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Review by miklos-da — APPROVED** [review]

**Review by andreaslochbihler-da — APPROVED** [review]
Looks good from the Canton perspective.

File: ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/Rejection.scala [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  val definiteAnswer: Boolean
+
+  def toDomainRejectionReason: domain.RejectionReason
+
+  def toStateV1RejectionReason: v1.RejectionReason =
+    toDomainRejectionReason.toParticipantStateRejectionReason
+}
+
+object Rejection {
+  val rejectionDomain = "com.daml.ledger"
```
[the remark this answers is not in the document]
Comment (SamirTalwar-DA) [reply ledger/sandbox-classic/src/main/scala/platform/sandbox/stores/ledger/Rejection.scala:28]: I shouldn't have committed this code; it's for v2 and I decided against including it in this PR. Deleting it.

**Review by fabiotudone-da — COMMENTED** [review]
How about adding unit tests for the conversions?

**Comment by SamirTalwar-DA** [comment]
@fabiotudone-da: Good call. I was being lazy because these conversions are temporary, but I think tests are still important because they'll force me to test the changes too.

File: ledger/participant-integration-api/src/test/suite/scala/platform/store/ConversionsSpec.scala [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+package com.daml.platform.store
+
+import com.daml.ledger.api.domain
+import com.daml.ledger.participant.state.{v1 => state}
+import com.daml.platform.store.Conversions._
+import org.scalatest.matchers.should.Matchers
+import org.scalatest.wordspec.AsyncWordSpec
+
+class ConversionsSpec extends AsyncWordSpec with Matchers {
```
Comment (SamirTalwar-DA) [inline ledger/participant-integration-api/src/test/suite/scala/platform/store/ConversionsSpec.scala:12]: @fabiotudone-da: Tests are here.

Comment (fabiotudone-da) [reply ledger/participant-integration-api/src/test/suite/scala/platform/store/ConversionsSpec.scala:12]: How about the `toStateV1RejectionReason` conversions?

**Pull request merged**